src/kerberos/mod.rs

```diff
@@ -173,29 +175,26 @@ impl Kerberos {
         }
     }
 
-    fn send(&self, data: &[u8]) -> Result<Vec<u8>> {
+    async fn send<'data>(&self, yield_point:&mut YieldPointLocal,data: &'data [u8]) -> Result<Vec<u8>> {
         if let Some((realm, kdc_url)) = self.get_kdc() {
             let protocol = NetworkProtocol::from_url_scheme(kdc_url.scheme()).ok_or_else(|| {
                 Error::new(
                     ErrorKind::InvalidParameter,
                     format!("Invalid protocol `{}` for KDC server", kdc_url.scheme()),
                 )
             })?;
-
-            if !self.config.network_client.is_protocol_supported(protocol) {
-                return Err(Error::new(
-                    ErrorKind::InvalidParameter,
-                    format!(
-                        "Network protocol `{}` is not supported by `{}` network client. Supported protocols are: {:?}",
-                        kdc_url.scheme(),
-                        self.config.network_client.name(),
-                        self.config.network_client.supported_protocols(),
-                    ),
-                ));
-            }
+            let request = NetworkRequest {
+                protocol,
+                url: kdc_url.clone(),
+                data:data.to_vec()
+            };
```

Comment:
This value is not used except in the `Tcp` branch

Reply:
updated